lte/cloud/go/services/subscriberdb/digest.go

```diff
@@ -52,9 +53,92 @@ func GetDigest(networkID string) (string, error) {
 	return digest, nil
 }
 
+// GetPerSubDigests generates a set of individual subscriber digests ordered by their IDs.
+func GetPerSubDigests(network string) ([]*lte_protos.SubscriberDigestByID, error) {
+	// HACK: apn resources digest is concatenated to every individual subscriber digest
+	// so that the entire set of per-sub digests would capture changes in apn resources
+	apnDigest, err := getApnResourcesDigest(network)
+	if err != nil {
+		glog.Errorf("Failed to generate digest for apn resources of network %+v: %+v", network, err)
+		apnDigest = ""
+	}
+
+	apnsByName, err := LoadApnsByName(network)
+	if err != nil {
+		return nil, err
+	}
+	perSubDigests := []*lte_protos.SubscriberDigestByID{}
+	token := ""
+	foundEmptyToken := false
+	for !foundEmptyToken {
+		// the resultant subProtos lists are already ordered by subscriber id due to the nature
+		// of LoadSubProtosPage
+		subProtos, nextToken, err := LoadSubProtosPage(0, token, network, apnsByName, lte_models.ApnResources{})
+		if err != nil {
+			return nil, err
+		}
+		for _, subProto := range subProtos {
+			digest, err := mproto.HashDeterministic(subProto)
+			if err != nil {
+				// Swallow errors related to a single subscriber digest to avoid affecting the rest
+				glog.Errorf("Failed to generate digest for subscriber %+v of network %+v: %+v", subProto.Sid.Id, network, err)
+				digest = ""
+			}
+			fullDigest := digest + apnDigest
+			perSubDigest := &lte_protos.SubscriberDigestByID{
+				Digest: &lte_protos.Digest{Md5Base64Digest: fullDigest},
+				Sid:    subProto.Sid,
+			}
+			perSubDigests = append(perSubDigests, perSubDigest)
+		}
+		foundEmptyToken = nextToken == ""
+		token = nextToken
+	}
+
+	return perSubDigests, nil
+}
+
+// GetPerSubDigestsDiff computes the changeset between two lists of per-subscriber digests,
```

Comment:
Great job on this fn, I was worried its impl would look messy but you made it nice and clear